tox.ini

```diff
@@ -1,27 +1,49 @@
 [tox]
 skipsdist = True
-envlist = py37-{lint,mypy}, docs
+envlist = py{34,35,36,37}-test, lint, py37-mypy, docs
 
 [travis]
 python =
   3.7: py37, docs
 
 [testenv]
 deps =
-  lint: pylint~=2.3.1
   mypy: mypy~=0.711
 
+setenv =
+  PYTHONPATH={toxinidir}/opentelemetry-api/src/
+  mypy: MYPYPATH={env:PYTHONPATH}
+
+changedir =
+  test: opentelemetry-api/tests
 
 commands =
-  py37-lint: pylint opentelemetry-api/src/opentelemetry/
   py37-mypy: mypy opentelemetry-api/src/opentelemetry/
+; For test code, we don't want to use the full mypy strictness, so we use its
+; default flags instead.
+  py37-mypy: mypy opentelemetry-api/tests/ --config-file=
+  test: python -m unittest discover
+
+[testenv:lint]
+deps =
+  pylint~=2.3.1
```

Comment:
Not directly related to this PR, do we want to put a specific version or we want to use the latest version?

Putting a specific version gives us the benefit of a more stable CI enironment (it is not perfect since even if we put a specific package version, it could introduce other package dependencies that are not fixed), in the cost that when we decide to move to a new version, we end up doing a big change (which means bigger risk) instead of doing small changes through our daily work.

Reply:
We use `~=` (https://www.python.org/dev/peps/pep-0440/#compatible-release) here, but in fact, I think using `=2.*,>=2.3.1` or `~=2.3` instead would be better. I would definitely pin the major version though.

Reply:
I'd be happy with any of these, from most- to least-conservative: `~=2.3.1`, `=2.*,>=2.3.1`,  `~=2.3`, `>=2.3.1`.

Since this is a test dependency and should only affect devs, I think there's actually a good argument for not pinning the major version and staying up to date with the latest versions in the course of regular dev work.

Reply:
Changed these to `~=2.3`.